### PR TITLE
Add wrappers for stat64 family calls

### PR DESCRIPTION
**Pull request by wangvsa**
### Description
This PR includes wrappers for stat64 family calls, including stat64, fstat64, __xstat64, __fxstat64 and __lxstat64.
Note: I changed the `unifyfs_file_attr_to_stat` function from a static function to a marco so it can handle both `struct stat` and `struct stat64` as the input parameter.

### Motivation and Context
stat64 family calls are used by many applications that need to access large files. They are used either explicitly or implicitly (e.g., through FILE_OFFSET_BITS=64 compile flags). See issue https://github.com/LLNL/UnifyFS/issues/769.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Testing (addition of new tests or update to current tests)
- [ ] Documentation (a change to man pages or other documentation)

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -267,10 +274,19 @@ AC_CHECK_FUNCS(fstatfs,[
...
 ],[])
+AC_CHECK_FUNCS(__xstat64,[
+    LINK_WRAPPERS+=",-wrap,__xstat64"
+],[])
 AC_CHECK_FUNCS(__fxstat,[
+    LINK_WRAPPERS+=",-wrap,__fxstat64"
+],[])
+AC_CHECK_FUNCS(__fxstat64,[
     LINK_WRAPPERS+=",-wrap,__fxstat"
 ],[])
```
Comment (CamStan) [inline configure.ac:291]: A lot of parentheses and brackets so I could be seeing things here, but it looks like the `__fxstat` and `__fxstat64` are swapped here. I.e., the wrap for `__fxstat64` is inside the `__fxstat` function and vice-versa.

```suggestion
AC_CHECK_FUNCS(__fxstat,[
    LINK_WRAPPERS+=",-wrap,__fxstat"
],[])
AC_CHECK_FUNCS(__fxstat64,[
    LINK_WRAPPERS+=",-wrap,__fxstat64"
],[])
```

File: common/src/unifyfs_meta.h [hunk 2]
```diff
@@ -263,40 +263,43 @@ int unifyfs_file_attr_update(int attr_op,
...
-         */
-        sb->st_nlink = fattr->is_laminated ? 1 : 0;
-
-        sb->st_atim = fattr->atime;
-        sb->st_mtim = fattr->mtime;
-        sb->st_ctim = fattr->ctime;
-    }
-}
+/*
+ * Convert UnifyFS file attr to sturct stat/stat64
```
Comment (CamStan) [inline common/src/unifyfs_meta.h:267]: Typo in the comment:
```suggestion
 * Convert UnifyFS file attr to struct stat/stat64
```

**Review by CamStan — CHANGES_REQUESTED** [review]
Thanks @wangvsa! Looks great! Just one typo and something that might be off in `configure.ac`

File: configure.ac (hunk 1 above)
Comment (wangvsa) [reply configure.ac:291]: Good catch, must be something off during the squash, will fix it.

File: common/src/unifyfs_meta.h (hunk 2 above)
Comment (wangvsa) [reply common/src/unifyfs_meta.h]: Fixed

**Pull request merged**